core/vm/environment.go

```diff
@@ -35,6 +37,7 @@ type (
 
 // Context provides the EVM with auxilary information. Once provided it shouldn't be modified.
 type Context struct {
+	context.Context
```

Comment:
I don't like this. The context should be passed into the function as a separate parameter.
Making it part of a struct is usually wrong. See [context documentation](https://golang.org/pkg/context/):

> Programs that use Contexts should follow these rules to keep interfaces consistent across packages and enable static analysis tools to check context propagation:
> 
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:

It's also misplaced in the VM Context struct it is supposed to contain info about the "blockchain context" in which execution is taking place. 

If you want to enable quitting via context.Context, please pass it explicitly.
For recursive calls, you'll probably need to make additional unexported versions
of `call`, `create`, etc. because they don't need to take the context when recursing.